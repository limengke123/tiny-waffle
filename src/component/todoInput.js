import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from '../style/component/todoInput.module.scss'

class TodoInput extends Component {
    static propTypes = {
        value: PropTypes.string,
        onComplete: PropTypes.func,
        onChange: PropTypes.func
    }

    static defaultProps = {
        value: '',
        onComplete() {},
        onChange() {}
    }

    handleKeyUp = e => {
        if (e.keyCode === 13) {
            const { onComplete } = this.props
            onComplete()
        }
    }

    render() {
        const { value, onChange } = this.props
        return (
            <div>
                <input
                    type="text"
                    value={value}
                    onChange={onChange}
                    onKeyUp={this.handleKeyUp}
                    className={styles['todo-input']}
                />
            </div>
        )
    }
}

export { TodoInput }
