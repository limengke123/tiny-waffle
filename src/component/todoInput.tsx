import React, { PureComponent } from 'react'
import styles from '../style/component/todoInput.module.scss'

export interface TodoInputProps {
    value: string
    onComplete: () => void
    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
    placeholder: string
}

export class TodoInput extends PureComponent<TodoInputProps, {}, {}> {
    static defaultProps = {
        value: '',
        onComplete() {},
        onChange() {},
        placeholder: ''
    }

    handleKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
        if (e.keyCode === 13) {
            const { onComplete } = this.props
            onComplete()
        }
    }

    render() {
        const { value, onChange, placeholder } = this.props
        return (
            <div>
                <input
                    type="text"
                    value={value}
                    onChange={onChange}
                    onKeyUp={this.handleKeyUp}
                    className={styles['todo-input']}
                    placeholder={placeholder}
                />
            </div>
        )
    }
}
