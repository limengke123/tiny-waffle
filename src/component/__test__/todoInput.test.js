import React from 'react'
import { mount, render } from 'enzyme'
import { TodoInput } from '../todoInput'

describe('todoInput', () => {
    it('should render correctly value', () => {
        const wrapper = render(<TodoInput />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('input').text()).toBe('')

        const wrapper1 = render(<TodoInput value="input测试文字" />)
        expect(wrapper1).toMatchSnapshot()
        // expect(wrapper1.find('input').text()).toBe('input测试文字')
    })
})
