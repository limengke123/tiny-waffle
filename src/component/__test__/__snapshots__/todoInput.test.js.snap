// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todoInput should render correctly value 1`] = `
<div>
  <input
    class="todo-input"
    placeholder=""
    type="text"
    value=""
  />
</div>
`;

exports[`todoInput should render correctly value 2`] = `
<div>
  <input
    class="todo-input"
    placeholder=""
    type="text"
    value="input测试文字"
  />
</div>
`;
